openedx/features/course_experience/tests/views/test_course_home.py

```diff
@@ -323,14 +324,14 @@ def test_non_live_course(self):
         )
         self.assertRedirects(response, expected_url)
 
-    @override_waffle_flag(CONTENT_TYPE_GATING_FLAG, True)
     @mock.patch.dict(settings.FEATURES, {'DISABLE_START_DATES': False})
     def test_expired_course(self):
         """
         Ensure that a user accessing an expired course sees a redirect to
         the student dashboard, not a 404.
         """
         three_years_ago = now() - timedelta(days=(365 * 3))
+        ContentTypeGatingConfig.objects.create(enabled=True, enabled_as_of=date(2010, 1, 1))
```

Comment:
this needs to be CourseDurationLimitConfig, this should fix your test

Reply:
Aha! That's why those tests were failing!